examples/benchmarks/train_resnet_imagenet1k.py

```diff
@@ -0,0 +1,320 @@
+parser.add_argument('--eval_batch_size', help='Validation dataloader per-device batch size', type=int, default=2048)
+
+# Model arguments
+parser.add_argument('--model_name',
+                    help='Name of the resnet model to train',
+                    default='resnet50',
+                    choices=['resnet18', 'resnet34', 'resnet50', 'resnet101', 'resnet152'])
+parser.add_argument('--loss_name',
+                    help='Name of the loss function to use for training',
+                    default='cross_entropy',
+                    choices=['cross_entropy', 'binary_cross_entropy'])
+
+# Optimizer arguments
+parser.add_argument('--learning_rate', help='Optimizer learning rate', type=float, default=2.048)
+parser.add_argument('--momentum', help='Optimizer momentum', type=float, default=0.875)
+parser.add_argument('--weight_decay', help='Optimizer weight decay', type=float, default=5.0e-4)
+
+# LR scheduler arguments
+parser.add_argument('--t_warmup',
+                    help='Duration of learning rate warmup specified by a Time string',
+                    type=Time.from_timestring,
+                    default='8ep')
+parser.add_argument('--t_max',
+                    help='Duration to cosine decay the learning rate',
+                    type=Time.from_timestring,
+                    default='1dur')
+parser.add_argument('--alpha_f', help='Learning rate multiplier to decay to', type=Time.from_timestring, default=0)
+
+# Save checkpoint arguments
+parser.add_argument('--save_checkpoint_dir',
+                    help='Directory to save checkpoints',
+                    type=str,
+                    default='checkpoint/{run_name}')
+parser.add_argument('--checkpoint_interval', help='Frequency to save checkpoints', type=str, default='1ep')
+
+# Load checkpoint arguments, assumes resuming the previous training run instead of fine-tuning
+parser.add_argument('--load_checkpoint_path', help='Path to the checkpoint to load', type=str)
+
+# Recipes
+parser.add_argument('--recipe_name',
+                    help='Either "mild", "medium" or "spicy" in order of increasing training time and accuracy',
+                    type=str,
+                    choices=['mild', 'medium', 'spicy'])
+
+# Logger parameters: progress bar logging is used by default
+# Only has Weights and Biases option to reduce the number of arguments. Other loggers can be substituted in the script
+parser.add_argument('--wandb_logger', help='Whether or not to log results to Weights and Biases', action='store_true')
+parser.add_argument('--wandb_entity', help='WandB entity name', type=str)
+parser.add_argument('--wandb_project', help='WandB project name', type=str)
+parser.add_argument('--wandb_run_name', help='WandB run name', type=str)
+
+# Trainer arguments
+parser.add_argument('--seed', help='Random seed', type=int, default=17)
+parser.add_argument('--grad_accum', help='Gradient accumulation either an int or "auto"', default='auto')
```

Comment:
Hard to fit in one line so maybe not worth it, but would be great to specify what the int options mean. Like, does grad accum = 0 turn it off (no extra batches get accumulated), or does grad accum = 1 turn it off (one batch per update)?

Reply:
I will remove this argument as well in favor of fixing `grad_accum='auto'` since there doesn't seem to be an easy way for arg parse to take two argument types? (I didn't dig too deep...)